src/Microsoft.ComponentDetection.Detectors/gradle/GradleComponentDetector.cs

```diff
@@ -45,8 +57,24 @@ protected override Task OnFileFoundAsync(ProcessRequest processRequest, IDiction
         return Task.CompletedTask;
     }
 
+    private List<string> ReadEnvVarStringList(IEnvironmentVariableService envVarService, string envVar)
+        => (envVarService.GetEnvironmentVariable(envVar) ?? string.Empty).Split(",", StringSplitOptions.RemoveEmptyEntries).ToList();
+
     private void ParseLockfile(ISingleFileComponentRecorder singleFileComponentRecorder, IComponentStream file)
     {
+        // Buildscript and Settings lockfiles are always development dependencies
+        var lockfileRelativePath = Path.GetRelativePath(this.CurrentScanRequest.SourceDirectory.FullName, file.Location);
+        var lockfileIsDevDependency = lockfileRelativePath.EndsWith("buildscript-gradle.lockfile") || lockfileRelativePath.EndsWith("settings-gradle.lockfile") || this.devLockfiles.Contains(lockfileRelativePath);
```

Comment:
You should just be able to compare the file names directly rather than `lockfileRelativePath.EndsWith` unless Gradle allows `abc-buildscript-gradle.lockfile` as valid names.